Source/FieldSolver/WarpXExternalEMFields.cpp

```diff
@@ -3,43 +3,53 @@
 #include "Utils/WarpXUtil.H"
 #include "Parser/WarpXParserWrapper.H"
 #include "Parser/GpuParser.H"
+#include "Utils/WarpXAlgorithmSelection.H"
 
 using namespace amrex;
 
+/**
+ * \brief which_variable determines which field component the external excitation is applied on
+ * which_variable = 0 : external field excitation applied to all three field components, E, B and H
+ * which_variable = 1 : external field excitation applied to E field only
+ * which_variable = 2 : external field excitation applied to B field only
+ * which_variable = 3 : external field excitation applied to H field only; this option is only valid when USE_LLG == TRUE
+ */
+
 void
-WarpX::ApplyExternalFieldExcitationOnGrid ()
+WarpX::ApplyExternalFieldExcitationOnGrid (std::string const& fieldtype) 
```

Comment:
I’m not so confident in the input variable I have in `WarpX::ApplyExternalFieldExcitationOnGrid (std::string const& fieldtype)`  , does the constant string input make sense?

Reply:
Its better to use
`int const fieldtype`